src/tasks/netSdk/NetSdkRunTaskProvider.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, CustomExecution, ProviderResult, Task, TaskDefinition, TaskScope } from "vscode";
+import { DockerPseudoterminal } from "../DockerPseudoterminal";
+import { DockerTaskProvider } from '../DockerTaskProvider';
+import { DockerTaskExecutionContext } from '../TaskHelper';
+import { NET_SDK_RUN_TASK_SYMBOL, NetSdkTaskHelper } from './NetSdkTaskHelper';
+
+const netSdkDebugTaskName = 'debug';
+export class NetSdkRunTaskProvider extends DockerTaskProvider {
+
+    public constructor(protected readonly helper: NetSdkTaskHelper) { super(NET_SDK_RUN_TASK_SYMBOL, undefined); }
+
+    provideTasks(token: CancellationToken): ProviderResult<Task[]> {
+
+        // we need to initialize a task first so we can pass it into `DockerPseudoterminal`
+        const task = new Task(
```

Comment:
This feels weird. Is there no way to use the same task?

Reply:
I tried a few ways to rewrite it but I wasn't able to...

Reply:
update: was able to do a little magic to avoid this w/ @bwateratmsft 's help